apps/nextjs-example/src/components/WalletSelector.tsx

```diff
@@ -35,7 +44,23 @@ import {
 } from "./ui/dropdown-menu";
 import { useToast } from "./ui/use-toast";
 
-export function WalletSelector() {
+export interface WalletSelectorProps {
```

Comment:
thoughts on this lives in the react package and each selector package imports it? looks like we keep reimplementing it on any selector ui package

Reply:
It would be pretty difficult to make a generic version of this one component. The Radix, Ant, and MUI implementation each use different styling solutions. Radix uses Tailwind, Ant uses vanilla CSS, and MUI uses Emotion (via the sx prop). To move this into `wallet-adapter-react`, this component would have to accept a `className` prop for every single element in the selector and it's still not clear to me how I would handle the `asChild` pattern I'm using to render the headless components as Radix, Ant, and MUI specific components. I think using a composition model with these headless components is a better approach than trying to write a component with dozens of props that tries to satisfy each implementation.

Reply:
oh I was only talking about the WalletSelectorProps interface

Reply:
Oh I see. I'll think about how we should share this. I could just drop this interface into the shared package, but it might a bit strange to export an interface that isn't tied to any specific implementation in the package. Maybe I'll add another utility to the shared package that consolidates all the below logic into a single function:

```ts
const {
  /** Wallets that use social login to create an account on the blockchain */
  aptosConnectWallets,
  /** Wallets that use traditional wallet extensions */
  otherWallets,
} = getAptosConnectWallets(wallets);

const {
  /** Wallets that are currently installed or loadable. */
  defaultWallets,
  /** Wallets that are NOT currently installed or loadable. */
  moreWallets,
} = partitionWallets(otherWallets);

if (sortDefaultWallets) defaultWallets.sort(sortDefaultWallets);
if (sortMoreWallets) moreWallets.sort(sortMoreWallets);
```

Reply:
I just added a `groupAndSortWallets` function to `wallet-adapter-core` that encapsulates the grouping and sorting logic while providing an interface that includes all of those sorting options that were previously copy and pasted across the wallet selector implementations.